process/sync/shardblock.go

```diff
@@ -546,6 +551,11 @@ func (boot *ShardBootstrap) requestMiniBlocks(hashes [][]byte) {
 	err := boot.miniBlocksResolver.RequestDataFromHashArray(hashes)
 	if err != nil {
 		log.Debug("RequestDataFromHashArray", "error", err.Error())
+		return
+	}
+
+	for _, hash := range hashes {
+		boot.requestedItemsHandler.Add(string(hash))
```

Comment:
For large ranges this also might prove overkill as sweep is done on each and every iteration

Reply:
refactored